example-workflows/github-deployments.yml

```diff
@@ -0,0 +1,79 @@
+---
+name: 'deploy'
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cloning repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Create deployment
+        id: create_deployment
+        uses: octokit/request-action@v2.x
+        with:
+          route: POST /repos/:repository/deployments
+          repository: ${{ github.repository }}
+          ref: ${{ github.sha }}
+          environment: production
```

Comment:
Is there a way to change the environment based on whether its a PR or not? Might be good to make this whole example something that uses review-apps when not master as well?

Reply:
I'll see what I can do 👍

Reply:
I've setup an example repo to play with the workflow for review apps, see https://github.com/badsyntax/dokku-github-actions-demo/pull/4

The problem with using environments for review apps is you can only have one deployment per environment.  So this needs some more thought. I'm still looking into this and I'll update this PR if/when I can find a solution for this.

Reply:
[The docs](https://docs.github.com/en/actions/reference/environments) mention:

> Running a workflow that references an environment that does not exist will create an environment with the referenced name.

So this could be the solution, although when i tried this without the environment existing, i was getting 404's, but I could have been missing something.

Reply:
Another approach is to use 1 environment (eg `review`) and set `auto_inactive` to false to keep existing deploys active. https://docs.github.com/en/rest/reference/repos#inactive-deployments

Reply:
I've made some good progress with this and GitHub Deploys are pretty awesome! We can use GitHub Deploys to request reviews before deployment, which is especially useful for review apps, as we might not want a deploy on every PR (as this open for abuse for public repos).

![Capture](https://user-images.githubusercontent.com/102141/105867848-7208cf80-5fed-11eb-87f6-2339bb1cdf2c.PNG)

I'm getting close to updating this PR.